### ISSUE TITLE
Limit how many branches and tags can be updated in a single push

### ISSUE THREAD

**Issue by JamieMagee** [ossf/allstar#231]
GitHub has a new feature[^1] to block potentially destructive git pushes. Currently the feature is in beta, and there is no official API to control this setting, but it appears to be set with a `POST` request to `https://github.com/{owner}/{repo}/settings/update_push_settings` with the body

```json
{
  "enable_max_pushes": true,
  "max_pushes_count": 123
}
```

[^1]: https://github.blog/changelog/2022-06-15-block-potentially-destructive-git-pushes/

**Comment by jeffmendoza** [comment]
Looks cool, thanks for sharing. Seems like we can easily support the "Fix" action for this across a whole org, and the "policy" would be the maximum allowed value. I assume a `GET` works on this endpoint for allstar to check. I'm wondering if we should wait a bit to get some confirmation on the API, hmm.

**Comment by JamieMagee** [comment]
Yeah, I would wait until this API makes it out of preview until adding support to allstar. It will likely land under the `https://api.github.com/repos/{owner}/{repo}` API like other repo settings.
